### PR TITLE
Feature chalk docs

### PR DESCRIPTION
**Pull request by Meiguro**
The long awaited documentation for all the new features. All the newly introduced color fields are documented, new elements Line and Radial, and new modules Platform and Feature. Two guides are added in order to give a clearer direction for developing on Pebble.js in light of supporting Chalk.

Once everything looks good, a CloudPebble update will be requested, and this will be merged. This will address #96, #116, #156 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,7 +150,248 @@ wind.add(textfield);
...
+
+In the examples above, mostly black text on white or light gray is used which has the most contrast. Try to maintain this amount of contrast with text. Using dark gray on light gray for example can be unreadable at certain angles in the sunlight or in darkly lit areas.
+
+## Feature Detection
+[Feature Detection]: #feature-detection
+
+Pebble.js provides the [Feature] module so that you may perform feature detection. This allows you change the presentation or behavior of your application based on the capabilities or characteristics of the current Pebble watch that the user is running your application with.
+
+### Using Feature
+[Using Feature]: #feature-vs-platform
```
Comment (cat-haines) [inline README.md]: should be #using-feature, #feature-vs-platform is defined below

File: README.md [hunk 2]
```diff
@@ -218,7 +462,167 @@ Wakeup.schedule(
...
+
+**NOTE:** [Window]s also have a [Window.size()] method which returns its size as a [Vector2]. Use [Window.size()] when possible.
+
+````js
+var res = Feature.resolution();
+console.log('Current display is ' + res.x + 'x' + res.y);
+````
+
+#### Feature.actionBarWidth()
+
```
Comment (cat-haines) [inline README.md]: needs a link

File: README.md [hunk 3]
```diff
@@ -218,7 +462,167 @@ Wakeup.schedule(
...
+var textAlign = Feature.round('center', 'left');
+
+if (Feature.round()) {
+  // Perform round display only operation
+  console.log('Round display');
+}
+````
+
+#### Feature.microphone([yes, no])
+
```
Comment (cat-haines) [inline README.md]: needs a link

File: README.md [hunk 4]
```diff
@@ -218,7 +462,167 @@ Wakeup.schedule(
...
+### Platform
+
+The Platform module allows you to determine the current platform runtime on the watch through its `Platform.version` method. This is to be used when the [Feature] module does not give enough ability to discern whether a feature exists or not.
+
+````js
+var Platform = require('platform');
+````
+
+#### Platform.version()
+
```
Comment (cat-haines) [inline README.md]: needs a link

File: README.md [hunk 5]
```diff
@@ -218,7 +462,167 @@ Wakeup.schedule(
...
 ````
 
+## Platform
+[Platform]: #platform
+
+`Platform` provides a module of the same name `Platform` and a feature detection module [Feature].
+
+
+### Platform
+
```
Comment (cat-haines) [inline README.md]: needs a link?

File: README.md [hunk 6]
```diff
@@ -165,7 +408,7 @@ Exporting is possible by modifying or setting `module.exports` within the requir
 
 ### Pebble
 
-The `Pebble` object from [PebbleKit JavaScript](https://developer.pebble.com/guides/pebble-apps/pebblekit-js/) is available as a global variable. Its usage is discouraged in Pebble.js, instead you should use the objects documented below who provide a cleaner object interface to the same functionalities.
+The `Pebble` object from [PebbleKit JavaScript](https://developer.pebble.com/guides/pebble-apps/pebblekit-js/) is available as a global variable. Its usage is discouraged in Pebble.js where there are equivalents. You should use the objects documented below that provide a cleaner object interface to the same functionalities. Use the `Pebble` object when there is no Pebble.js alternative.
 
 ### window -- browser
```
Comment (cat-haines) [inline README.md]: link?

File: README.md [hunk 7]
```diff
@@ -165,7 +408,7 @@ Exporting is possible by modifying or setting `module.exports` within the requir
 
 ### Pebble
 
-The `Pebble` object from [PebbleKit JavaScript](https://developer.pebble.com/guides/pebble-apps/pebblekit-js/) is available as a global variable. Its usage is discouraged in Pebble.js, instead you should use the objects documented below who provide a cleaner object interface to the same functionalities.
+The `Pebble` object from [PebbleKit JavaScript](https://developer.pebble.com/guides/pebble-apps/pebblekit-js/) is available as a global variable. Its usage is discouraged in Pebble.js where there are equivalents. You should use the objects documented below that provide a cleaner object interface to the same functionalities. Use the `Pebble` object when there is no Pebble.js alternative.
```
Comment (cat-haines) [inline README.md]: "Its usage is discouraged in Pebble.js where there are equivalents."

What are the equivalents?

File: README.md [hunk 8]
```diff
@@ -292,13 +697,13 @@ document.location = 'pebblejs://close#' + encodeURIComponent(JSON.stringify(opti
 
 #### Settings.option(field, value)
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 9]
```diff
@@ -326,11 +731,16 @@ Settings.option({
 
 #### Settings.option()
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 10]
```diff
@@ -326,11 +731,16 @@ Settings.option({
...
 var options = Settings.option();
 console.log(JSON.stringify(options));
+
+options.counter = (options.counter || 0) + 1;
+
+// Modifications are not saved until `Settings.option` is called as a setter
+Settings.option(options);
 ````
 
 #### Settings.data
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 11]
```diff
@@ -341,13 +751,13 @@ While localStorage is still accessible, it is recommended to use `Settings.data`
 
 #### Settings.data(field, value)
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 12]
```diff
@@ -375,18 +785,25 @@ Settings.data({
 
 #### Settings.data()
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 13]
```diff
@@ -538,26 +959,77 @@ Pebble.js provides three types of Windows:
...
+});
+
+// Set a single status property
+card.status('separator', 'none');
+
+// Disable the status bar
+card.status(false);
+````
+
 #### Window.show()
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 14]
```diff
@@ -630,7 +1102,7 @@ wind.on('hide', function() {
 
 #### Window.action(actionDef)
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 15]
```diff
@@ -639,17 +1111,71 @@ card.action({
...
 ````
 
+To disable the action bar after enabling it, `false` can be passed in place of an `actionDef`.
+
+````js
+// Disable the action bar
+card.action(false);
+````
+
 #### Window.action(field, value)
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 16]
```diff
@@ -639,17 +1111,71 @@ card.action({
...
 
-You may also call `Window.action` with two arguments, `field` and `value`, to set specific fields of the window's `action` propery. `field` is a string refering to the [actionDef] property to change. `value` is the new property value to set.
+`Window.action` can also be called with two arguments, `field` and `value`, to set specific fields of the window's `action` property. `field` is the name of a [Window actionDef] property as a string and `value` is the new property value.
+
+````js
+card.action('select', 'images/action_icon_checkmark.png');
+````
+
+#### Window.status(statusDef)
+
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 17]
```diff
@@ -639,17 +1111,71 @@ card.action({
...
+Similarly, `true` can be used as a [Window statusDef] to represent a `statusDef` with all default properties.
+
+````js
+var card = new UI.Card({ status: true });
+card.show();
+````
+
+#### Window.status(field, value)
+
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 18]
```diff
@@ -639,17 +1111,71 @@ card.action({
...
+var wind = new UI.Window({ status: true });
+
+var size = wind.size();
+var rect = new UI.Rect({ size: new Vector2(size.x / 4, size.y / 4) });
+wind.add(rect);
+
+wind.show();
+````
 
 ### Window (dynamic)
```
Comment (cat-haines) [inline README.md]: link

File: README.md [hunk 19]
```diff
@@ -949,14 +1502,50 @@ Accessor to the `borderColor` property. See [Element].
...
+  position2: new Vector2(72, 84),
+  strokeColor: 'white',
+});
+
+wind.add(line);
+wind.show();
+````
+
+#### Line.position2(position)
+
```
Comment (cat-haines) [inline README.md]: link?

File: README.md [hunk 20]
```diff
@@ -148,7 +150,248 @@ wind.add(textfield);
...
+});
+
+card.show();
+````
+
+Now, if it is necessary to separate the different logic in setting up the card, the individual units can be implemented in separate functions without anything unintentionally mixing the logic together. [Feature] is provided as a module, so it is always available where you decide to move your logic.
+
+The two examples consist of logic confined into one line, but if each line was instead large blocks of logic with the `isAplite` boolean used throughout, the issue is more apparent, hence the recommendation to use [Feature] detection. Of course, for capabilities or characteristics that [Feature] is unable to allow you to discern, use [Platform].
+
+**NOTE:** [Feature] underneath the hood does not actually perform feature detection. The benefit of using [Feature] however is that the detection logic is abstracted out of your application, so if any platform is changed or introduced, Pebble.js will update and your application will automatically adapt in the case that no new capabilities or characteristics are introduced.
```
Comment (cat-haines) [inline README.md]: "[Feature] underneath the hood does not actually perform feature detection. "

What does this mean, and why do I care about it as a developer?

**Comment by cat-haines** [comment]
You seem to use links inconsistently in your section headers (I marked a bunch, not all) - not certain if intentional or not...

**Comment by Meiguro** [comment]
Yes, intentional, the links are only defined if they are necessary. When the header has no strange symbols in it no anchor tag is necessary and it automatically works, but for some odd reason when there are parens or square brackets it's necessary to define your own, or possibly I've not figured out the escape sequence.

File: README.md (hunk 20 above)
Comment (Meiguro) [inline README.md]: I was actually on the fence with this one, it was included for completeness. You're right, it is confusing and unnecessary to document, I'll remove it. What it means, however, is that actual feature detection is when the feature is directly tested for, such as determining if there is a microphone by attempting to use it, but that was removed due to code size issues, so it probably doesn't help pebble.js' cause to mention real feature detection.

**Comment by Meiguro** [comment]
Thanks for the quick look over! I updated the docs per the comments. I want to eventually add anchors for all headers, but I'm unsure if that's the right way to use Flatdoc at the moment. When I figure out more, I'll fix them all up.

One of my main concerns is if I ordered the guides correctly. I kept Using Images and Using Fonts first since they're short. The two new guides are much longer, but I put the one that most users would find more immediately useful first, which I assume is Using Color. The one gotcha is that it uses `Feature` before they even read the Feature Detection guide, so I was unsure about that order. Maybe that makes sense since it just subtly introduces `Feature.color` and `Feature.round` in the examples, and they can read `Feature Detection` if they want a fuller understanding since they are actually simple functions anyway.

**Pull request merged**